src/client/resources/web/js/chat.mjs

```diff
@@ -4,98 +4,418 @@
 
+// Used to keep track of messages already shown. To prevent possible duplication on server join.
+/** @type {Set<string>} */
+const displayedMessageIds = new Set();
+
+/**
+ * Server information and related methods.
+ * @type {{
+ *   name: string | undefined;
+ *   id: string | undefined;
+ *   baseTitle: string;
+ *   update: (name: string , id: string) => void;
+ *   clear: () => void;
+ *   getId: () => string | undefined;
+ *   getName: () => string | undefined;
+ * }}
+ */
+const serverInfo = {
+    name: undefined,
+    id: undefined,
+    baseTitle: document.title, // Store the page title on load so we can manipulate it based on events and always restore it.
+
+    /**
+     * Updates server information and UI elements.
+     * @param {string} name - The server's name.
+     * @param {string} id - The server's identifier.
+     */
+    update(name, id) {
+        if (!name || !id) {
+            console.error('Invalid server information: Both name and id must be provided.');
+            return;
+        }
+
+        this.name = name;
+        this.id = id;
+
+        // Update the page title
+        document.title = `${this.baseTitle} - ${name}`;
+
+        // Update the status element
+        if (serverNameElement) {
+            serverNameElement.textContent = name;
+        }
+    },
+
+    /**
+     * Clears the current server information from both variables and UI.
+     */
+    clear() {
+        this.name = undefined;
+        this.id = undefined;
+        document.title = this.baseTitle;
+        if (serverNameElement) {
+            serverNameElement.textContent = 'No server';
+        }
+
+    },
+
+    /**
+     * Retrieves the server's ID.
+     * @returns {string | undefined} The server's ID.
+     */
+    getId() {
+        return this.id;
+    },
+
+    /**
+     * Retrieves the server's name.
+     * @returns {string | undefined} The server's name.
+     */
+    getName() {
+        return this.name;
+    }
+};
+
+
+/**
+ * ======================
+ *  HTML elements
+ * ======================
+ */
+
+const statusContainerElement = /** @type {HTMLDivElement | null} */ (document.getElementById('status'));
+const statusTextElement = /** @type {HTMLSpanElement | null} */ (document.querySelector('#status .connection-status'));
+const serverNameElement = /** @type {HTMLSpanElement | null} */ (document.querySelector('#status .server-name'));
+
+const messagesElement = /** @type {HTMLDivElement | null} */ (document.getElementById('messages'));
+const loadMoreContainerElement = /** @type {HTMLDivElement | null} */ (document.getElementById('load-more-container'));
+const loadMoreButtonElement = /** @type {HTMLDivElement | null} */ (document.getElementById('load-more-button'));
+
+const chatInputElement = /** @type {HTMLTextAreaElement | null} */ (document.getElementById('message-input'));
+const messageSendButtonElement = /** @type {HTMLTextAreaElement | null} */ (document.getElementById('message-send-button'));
```

Comment:
We could assert these exist right here and then the variables would be non-null.

Reply:
Ah. do you suggest a not null operator at the end `!`?  Or maybe wrap them in an assertion method that throws an error?

Reply:
I do like runtime validation, so I am going with 

```
/**
 * Gets element based on selector. Throws error if element is null.
 * @template {Element} T
 * @param {string} selector
 * @returns {T}
 */
function querySelectorWithAssertion(selector) {
    const element = document.querySelector(selector);
    if (!element) {
        throw new Error(`Required DOM element not found: ${selector}`);
    }
    return /** @type {T} */ (element);
}

const statusContainerElement = querySelectorWithAssertion('#status');
```